wifiphisher/extensions/handshakeverify.py

```diff
@@ -0,0 +1,299 @@
+        self._eapols = deque(self._store_eapols + list(self._eapols))
+        # reset the _store_eapols
+        self._store_eapols = []
+        # if captured the handshake but not done return fail
+        if self._is_captured:
+            return 'fail'
+
+    def get_packet(self, packet):
+        """
+        Process the Dot11 packets and verifiy it is a valid
+        eapol frames in a 80211 fourway handshake
+        :param self: Fourway object
+        :param packet: A scapy.layers.RadioTap object
+        :type self: Fourway
+        :type packet: scapy.layers.RadioTap
+        :return: empty list
+        :rtype: list
+        ..note: In this extension we don't need to send the packets
+        to the extension manager.
+        """
+
+        # append the capture of user first:
+        if self._is_first and self._data.args.handshake_capture:
+            pkts = dot11.rdpcap(self._data.args.handshake_capture)
+            for pkt in pkts:
+                if self.is_valid_handshake_frame(pkt):
+                    self._eapols.append(pkt)
+            self._is_first = False
+
+        # check if verification is done
+        if self._is_done != DONE:
+            # append to list if this is the key frame
+            if self.is_valid_handshake_frame(packet):
+                self._eapols.append(packet)
+
+        num_of_frames = len(self._eapols)
+        for index in range(num_of_frames):
+            if num_of_frames - index > 3:
+                ap_bssid = self._data.target_ap_bssid
+                # from AP to STA
+                msg1 = self._eapols[0]
+                # from STA to AP
+                msg2 = self._eapols[1]
+                # from AP to STA
+                msg3 = self._eapols[2]
+                # from STA to AP
+                msg4 = self._eapols[3]
+
+                if msg1.addr2 == ap_bssid and\
+                        msg3.addr2 == ap_bssid and\
+                        msg2.addr1 == ap_bssid and\
+                        msg4.addr1 == ap_bssid:
+                    self._is_captured = True
+            else:
+                break
+
+        return [["*"], []]
+
+    def send_output(self):
+        """
+        Send the output the extension manager
+        :param self: A Fourway object.
+        :type self: Fourway
+        :return: A list with the password checking information
+        :rtype: list
+        """
+
+        ret_info = []
+        pw_str = "ESSID: {0}".format(
+            self._data.target_ap_essid)
+
+        # handshake has been captured but verify fail
+        if self._is_captured and self._is_done == FAIL:
+            ret_info = ["WPA HANDSHAKE CAPTURED - " + pw_str +
+                        " NOT CORRECT!"]
+        # handshake has been captured and wait for victim to
+        # type credentials
+        elif self._is_captured and self._is_done == NOT_YET:
+            ret_info = ["WPA HANDSHAKE CAPTURED - " + pw_str]
```

Comment:
What I'm saying is that In that statement you are returning to the user "WPA HANDSHAKE CAPTURED". However, it would be more descriptive to say "PSK Captured".